yt/geometry/coordinates/cartesian_coordinates.py

```diff
@@ -323,11 +324,20 @@ def _ortho_pixelize(
         # We should be using fcoords
         field = data_source._determine_fields(field)[0]
         finfo = data_source.ds.field_info[field]
-        period = self.period[:2].copy()  # dummy here
-        period[0] = self.period[self.x_axis[dim]]
-        period[1] = self.period[self.y_axis[dim]]
-        if hasattr(period, "in_units"):
-            period = period.in_units("code_length").d
+        # some coordinate handlers use only projection-plane periods,
+        # others need all box periods.
+        period2 = self.period[:2].copy()  # dummy here
+        period2[0] = self.period[self.x_axis[dim]]
+        period2[1] = self.period[self.y_axis[dim]]
+        period3 = self.period[:].copy()  # dummy here
+        period3[0] = self.period[self.x_axis[dim]]
+        period3[1] = self.period[self.y_axis[dim]]
+        zax = list({0, 1, 2} - {self.x_axis[dim], self.y_axis[dim]})[0]
+        period3[2] = self.period[zax]
+        if hasattr(period2, "in_units"):
+            period2 = period2.in_units("code_length").d
+        if hasattr(period3, "in_units"):
+            period3 = period3.in_units("code_length").d
```

Comment:
What is period3 and period2? What's the reasoning for these variable names? I can see before period was the axis that wasn't being projected..? But 3 is not a valid axis index...

Reply:
`period2` is a 2-element iterable, which only contains info about the period in the directions perpendicular to the line of sight, `period3` is a 3-element iterable, which also indicates whether the volume is periodic along the line-of-sight direction. Basically, `period3` is what I use in the SPH backend functions I updated, but `period2` is what backends for other simulation types need. The old SPH versions used `period2` as well, so I basically 'split' this variable into a version for each backend.